meteor/server/lib/customPublication/observerGroup.ts

```diff
@@ -0,0 +1,130 @@
+import { ManualPromise, createManualPromise } from '@sofie-automation/corelib/dist/lib'
+import { Meteor } from 'meteor/meteor'
+import { waitForPromise } from '../../../lib/lib'
+
+export interface ReactiveMongoObserverGroupHandle extends Meteor.LiveQueryHandle {
+	/**
+	 * Restart the observers inside this group
+	 */
+	restart(): void
+}
+
+/**
+ * Helper to trigger reactivity inside of an OptimisedObserver whenever one of the other Mongo observers changes
+ * Note: care needs to be taken when using this, as Mongo observers call `added` for every document when they start. It is very easy to form an infinite loop of observer invalidations
+ * @param generator Function to generate the `Meteor.LiveQueryHandle`s
+ * @returns Handle to stop and restart the observer group
+ */
+export function ReactiveMongoObserverGroup(
```

Comment:
You can blame me for this one, this came from my not-yet opened PR which this was based on.

Yeah it probably could be a class, I think the style of writing here was rather influenced by writing react functions. But I feel like making it a class will cause it to be clunkier to use once it has to stop using fibers (as it will need to return a Promise), which was incredibly lazy of me to write it with fibers...